site/posts/writing-multiple-netcdf-files-in-parallel-with-xarray-and-dask.ipynb

```diff
@@ -0,0 +1,1560 @@
+      "text/plain": [
+       "<IPython.core.display.Image object>"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "result.Tair.data.visualize()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Our resulting xarray dataset has the same dimensions and the same number of chunks as our input dataset."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Create a helper function to split a dataset into sub-datasets\n",
+    "\n",
+    "Our main objective is to save this resulting dataset into multiple netCDF files by saving each chunk in its own netCDF file. To accomplish this, we need to create a function that allows us to split a Dataset into sub-datasets for each chunk. The following code snippet was copied from this [comment](https://github.com/pydata/xarray/issues/1093#issuecomment-259213382) by [Stephan Hoyer](https://github.com/shoyer)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import itertools\n",
+    "def split_by_chunks(dataset):\n",
+    "    chunk_slices = {}\n",
+    "    for dim, chunks in dataset.chunks.items():\n",
+    "        slices = []\n",
+    "        start = 0\n",
+    "        for chunk in chunks:\n",
+    "            if start >= dataset.sizes[dim]:\n",
+    "                break\n",
+    "            stop = start + chunk\n",
+    "            slices.append(slice(start, stop))\n",
+    "            start = stop\n",
+    "        chunk_slices[dim] = slices\n",
+    "    for slices in itertools.product(*chunk_slices.values()):\n",
+    "        selection = dict(zip(chunk_slices.keys(), slices))\n",
+    "        yield (selection, dataset[selection])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "3\n"
+     ]
+    }
+   ],
+   "source": [
+    "blocks = list(split_by_chunks(result))\n",
+    "print(len(blocks)) # confirm that we have three items corresponding to our three chunks"
```

Comment:
This code comment:

> `# confirm that we have three items corresponding to our three chunks`

Should be removed and a new *markdown* cell added above this cell with the text:

```
Let's confirm that we have three items corresponding to our three chunks:
```